CMakeLists.txt

```
@@ -8,23 +8,28 @@ project(simdutf
 
 set(SIMDUTF_LIB_VERSION "1.0.1" CACHE STRING "simdutf library version")
 set(SIMDUTF_LIB_SOVERSION "1" CACHE STRING "simdutf library soversion")
+option(SIMDUTF_BENCHMARKS "Whether the benchmarks are included as part of the CMake Build." ON)
 
 include(GNUInstallDirs)
+include(CTest)
 include(cmake/simdutf-flags.cmake)
 
 set(SIMDUTF_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
 
-enable_testing()
 add_subdirectory(src)
-add_subdirectory(tests)
 add_subdirectory(singleheader)
-
-if(CMAKE_CXX_COMPILER_ID MATCHES GNU AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 8.0)
-  message(STATUS "The benchmark tool requires GCC 8.0 or better.")
-else()
-  add_subdirectory(benchmarks)
+if (BUILD_TESTING)
+  enable_testing()
+  add_subdirectory(tests)
 endif()
 
+if (SIMDUTF_BENCHMARKS)
+  if(CMAKE_CXX_COMPILER_ID MATCHES GNU AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 8.0)
```

```suggestion
if(SIMDUTF_BENCHMARKS AND CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
  if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "8.0")
```

[Docs](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER_ID.html). There is no compiler other than GCC with a compiler ID containing `GNU`, so regex is not really necessary here.

The message also suggests that benchmarking is a GCC only thing, so the conditional is lifted out. Ideally, `SIMDUTF_BENCHMARKS` would be a dependent option to begin with.

---

It looks like cmake dependent options do not support full conditional syntax until CMake 3.22, so I'm not sure I can use that here.

---

You can work around that with a variable like so: https://github.com/friendlyanon/SFML/blob/feature/cmake-rework/cmake/SFMLConfig.cmake.in#L41-L42

---

I'm going to avoid complicating this PR much further, and let someone else add both the pre-CMake 3.22 infrastructure and the changes to all the places in the CMake file they think are relevant.